### PR TITLE
Get rid of replaying an empty event list

### PR DESCRIPTION
**Pull request by dmytro-grankin**
Previously `AggregateRepository.loadOrCreate(id)` forced an aggregates with an empty event history to replay an events. Such a behavior was redundant and led to validation of an initial aggregate state. Because an initial aggregate state is not valid, the validation crashed a program. This PR fixes the problem.

Summary:
- `AggregateStorage.read()` now returns an absent `AggregateStateRecord`, if an aggregate history is empty.
- `AggregateRepository` now checks validity of an `AggregateStateRecord` instance.

### CODE REVIEW

**Comment by dmytro-grankin** [comment]
@armiol PTAL.

File: server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java [hunk 1]
```diff
@@ -79,8 +77,8 @@
  * @author Alexander Yevsyukov
  */
 public abstract class AggregateRepository<I, A extends Aggregate<I, ?, ?>>
-                extends Repository<I, A>
-                implements CommandDispatcher {
+        extends Repository<I, A>
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java]: Please revert the formatting changes.

File: server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java [hunk 2]
```diff
@@ -355,12 +355,15 @@ public void store(A aggregate) {
         return result;
     }
 
-    private static <I> IllegalStateException unableToLoadEvents(I id) {
-        final String errMsg = format(
-                "Unable to load events for the aggregate with ID: %s",
-                idToString(id)
-        );
-        throw new IllegalStateException(errMsg);
+    private static void checkAggregateStateRecord(AggregateStateRecord aggregateStateRecord) {
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java]: Please document this method.

File: server/src/main/java/org/spine3/server/aggregate/AggregateStorage.java [hunk 3]
```diff
@@ -56,6 +56,14 @@ protected AggregateStorage(boolean multitenant) {
         super(multitenant);
     }
 
+    /**
+     * Reads a {@link AggregateStateRecord} by the passed ID.
+     *
+     * @param aggregateId the ID of the record to read
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/aggregate/AggregateStorage.java]: This description is not formatted.

I also suspect it has been copied from its parent, so you might use `{@inheritDoc}` instead.

File: server/src/main/java/org/spine3/server/aggregate/AggregateStorage.java [hunk 4]
```diff
@@ -85,6 +93,10 @@ protected AggregateStorage(boolean multitenant) {
             }
         }
 
+        if (snapshot == null && history.isEmpty()) {
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/aggregate/AggregateStorage.java]: This should go before iterating over `historyBackward`.

File: server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java (hunk 2 above)
Comment (armiol) [inline server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java]: Also, I'd suggest testing this behaviour by using a mock `AggregateStorage`, which returns an invalid `Optional<AggregateStateRecord>` value upon `read(id)`.

File: server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java (hunk 1 above)
Comment (dmytro-grankin) [inline server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java]: Done.

File: server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java (hunk 2 above)
Comment (dmytro-grankin) [inline server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java]: Done.

Comment (dmytro-grankin) [inline server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java]: `AggregateRepositoryShould#reject_invalid_AggregateStateRecord_instance()` tests it.

File: server/src/main/java/org/spine3/server/aggregate/AggregateStorage.java (hunk 3 above)
Comment (dmytro-grankin) [inline server/src/main/java/org/spine3/server/aggregate/AggregateStorage.java]: Done.

File: server/src/main/java/org/spine3/server/aggregate/AggregateStorage.java (hunk 4 above)
Comment (dmytro-grankin) [inline server/src/main/java/org/spine3/server/aggregate/AggregateStorage.java]: Done.

**Comment by dmytro-grankin** [comment]
@armiol PTAL.

**Pull request merged**